fontforge/splinestroke.c

```diff
@@ -708,7 +708,7 @@ return;		/* Essentially colinear */ /* Won't be perfect because control points l
 	    p->needs_point_left = p->needs_point_right = false;
 	    p->left_hidden = bends_left;
 	    p->right_hidden = !bends_left;
-	    if ( rot.x<=fabs(diff_angle.x) ) {
+	    if ( rot.x<=diff_angle.x || (diff_angle.x <= -1 && rot.x <= -0.999999) ) { /* close enough */
```

Comment:
I have not read all of the code carefully, but this seems like an asymmetric fix to me. In particular, it seems odd to me that the negative case have a wider threshold than the positive case.